### DISCUSSION TITLE
tproxy 生命周期钩子可执行文件运行找不到(error when executing the transparent pre start hook * no such file)

### DISCUSSION THREAD

**Discussion by Ryan-Xu** [v2rayA/v2rayA#1458] [Q&A]
透明代理方式： tproxy
tproxy-hook.sh 文件存在且root权限可执行
启动参数配置：
ExecStart=/usr/bin/v2raya --log-disable-timestamp  --transparent-hook /etc/v2raya/tproxy-hook.sh

日志输出：
2024/07/28 00:02:46.820  [I] [io.go:431]  V2Ray 5.16.1 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.22.2 linux/amd64)
2024/07/28 00:02:46.820  [I] [io.go:431]  A unified platform for anti-censorship.
2024/07/28 00:02:46.938  [I] [processManager.go:217]  Execute the transparent pre start hook: /etc/v2raya/tproxy-hook.sh
2024/07/28 00:02:46.939  [E] [main.go:30]  failed to start v2ray-core: error when executing the transparent pre start hook: fork/exec /etc/v2raya/tproxy-hook.sh: no such file or directory
2024/07/28 00:02:46.948  [A] [index.go:116]  v2rayA is listening at http://127.0.0.1:2017

**Comment by eronez** [comment]
"/etc/v2raya/tproxy-hook.sh: no such file or directory"
Check location, permission, filename etc.
Execute command in terminal until it works.

**Comment by manorjust** [comment]
openwrt吗？
#!/bin/bash
改成
#!/bin/sh

**Reply by sungamma** [reply]
按此方法解决问题，把脚本里面的开头改成#!/bin/sh。用的docker版本

**Comment by sungamma** [comment]
一样的问题

**Reply by sungamma** [reply]
2025/01/23 02:14:43.431 [I] [processManager.go:220]  Execute the transparent pre start hook: [/etc/v2raya/tproxy-hook.sh --transparent-type=redirect --stage=pre-start --v2raya-confdir=/etc/v2raya]
2025/01/23 02:14:43.432 [E] [main.go:30]  failed to start v2ray-core: error when executing the transparent pre start hook: fork/exec /etc/v2raya/tproxy-hook.sh: no such file or directory
